enigma/src/main/java/cuchaz/enigma/translation/mapping/EntryResolver.java

```diff
@@ -21,7 +22,10 @@ default <E extends Entry<?>, C extends Entry<?>> Collection<EntryReference<E, C>
 
 		if (reference.context != null) {
 			Collection<C> context = resolveEntry(reference.context, strategy);
-			return Streams.zip(entry.stream(), context.stream(), (e, c) -> new EntryReference<>(e, c, reference)).toList();
+			List<EntryReference<E, C>> result = new ArrayList<>(entry.size());
+			new ZippingIterator<>(entry.iterator(), context.iterator(), (e, c) -> new EntryReference<>(e, c, reference))
+					.forEachRemaining(result::add);
```

Comment:
Rather than overcomplicating things and adding a new iterator, especially since this is only used to add to a `List` you could write this using a regular loop.